packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -268,9 +268,26 @@ module.exports = (
       if (cachedHeadings) {
         return cachedHeadings
       } else {
+        const getHeadingID = heading => {
+          const data = heading.data
+          if (data) {
+            if (data.id) return data.id
+            if (data.htmlAttributes && data.htmlAttributes.id) {
+              return data.htmlAttributes.id
+            }
+
+            if (data.hProperties && data.hProperties.id) {
+              return data.hProperties.id
+            }
+          }
+
+          return null
+        }
```

Comment:
Can we move this function outside? with this being here, the function will be constructed many times, and the function itself doesn't rely on variables in the scope of this one - so it can be defined as standalone "pure" function